src/hotspot/os/linux/os_linux.cpp

```diff
@@ -4253,11 +4253,16 @@ size_t os::vm_min_address() {
   static size_t value = 0;
   if (value == 0) {
     assert(is_aligned(_vm_min_address_default, os::vm_allocation_granularity()), "Sanity");
-    FILE* f = fopen("/proc/sys/vm/mmap_min_addr", "r");
-    if (fscanf(f, "%zu", &value) != 1) {
+    FILE* f = os::fopen("/proc/sys/vm/mmap_min_addr", "r");
+    if (f != nullptr) {
+      if (fscanf(f, "%zu", &value) != 1) {
```

Comment:
I think you can simplify the whole thing by only making this one change:
```
if (f != nullptr && fscanf(f, "%zu", &value) != 1) {
```
the "assignment" in the null case will be handled by the existing:
```
value = MAX2(_vm_min_address_default, value);
```

Reply:
Changed, but I'm not sure if the result is more simple. I still need a proper check for `f != nullptr` before `fclose`.

Reply:
Ah! I missed that.

Reply:
The existing code is overly complex. I think it can reduce to just:
```
if (f != nullptr) {
  fscanf(f, "%zu", &value);
  fclose(f);
}
value = MAX2(_vm_min_address_default, value);
```
though perhaps that makes an assumption that `value` is untouched if the conversion fails? I couldn't see anything definitive about that in the spec for it.
```

Reply:
I couldn't find any explicit statements in the spec either, but chances are good that it will just work. I'll make the change.

Reply:
well, it triggers a warning:
```
warning: ignoring return value of 'int fscanf(FILE*, const char*, ...)', declared with attribute warn_unused_result [-Wunused-result]
```
I'm not a big fan of suppressing warnings. Can we keep the current version?